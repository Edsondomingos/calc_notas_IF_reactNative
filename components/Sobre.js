import * as React from 'react';
import { useState } from 'react';
import { Text, View, StyleSheet, TouchableHighlight } from 'react-native';
import Styles from '../assets/styleSobre'

export default function Sobre(){
  return (
    <View style={Styles.container} testID='btnview' >
      <Text style={Styles.texto}>Bem vindo a calculadora de notas do IFRN</Text>
            <Text>para o calculo, usa-se média ponderada. </Text>
            <Text style={Styles.texto}>Caso esteja no ensino medio ou tecnico, possuindo 4 notas, a duas primeiras notas tem peso 2 e as duas ultimas peso 3. Com a soma das notas depois do calculo do peso, dividido pelo peso total, que no caso seria a soma de todos os pesos, resultando em 10, ou seja:
      </Text>
            
      <Text style={Styles.texto}>((B1 * 2) + (B2 * 2) + (B3 * 3) + (B4 * 3)) / 10</Text>

      <Text style={Styles.texto}>
            Para os cursos superiores, segue a mesma lógica, mas com apenas duas notas bimestrais, com peso total de 5, ou seja:
      </Text>

        <Text style={Styles.texto}>((B1 * 2) + (B2 * 3)) / 5 </Text>
      
    </View>
  )
}