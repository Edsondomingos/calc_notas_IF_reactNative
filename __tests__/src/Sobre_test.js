import renderer from 'react-test-renderer';
import { render, fireEvent } from "@testing-library/react-native";
import Sobre from '../../components/Sobre'
jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");

describe('Teste de Snapshot', () => {
    it('Snapshot de Sobre', () => {
        const tela = renderer.create(<Sobre />).toJSON();
        expect(tela).toMatchSnapshot();
    })
})

describe("Teste de inicialização e Exibição", ()=>{
    it("Component view na Sobre", ()=>{
        const { getByText, getByTestId } = render(<Sobre />);

        expect(getByText("Bem vindo a calculadora de notas do IFRN")).toBeTruthy();
        expect(getByTestId("btnview")).toBeTruthy();
    })
})
