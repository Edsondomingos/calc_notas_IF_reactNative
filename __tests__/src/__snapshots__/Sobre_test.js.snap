// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de Snapshot Snapshot de Sobre 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 10,
    }
  }
  testID="btnview"
>
  <Text
    style={
      Object {
        "fontSize": 15,
        "marginVertical": 10,
      }
    }
  >
    Bem vindo a calculadora de notas do IFRN
  </Text>
  <Text>
    para o calculo, usa-se média ponderada. 
  </Text>
  <Text
    style={
      Object {
        "fontSize": 15,
        "marginVertical": 10,
      }
    }
  >
    Caso esteja no ensino medio ou tecnico, possuindo 4 notas, a duas primeiras notas tem peso 2 e as duas ultimas peso 3. Com a soma das notas depois do calculo do peso, dividido pelo peso total, que no caso seria a soma de todos os pesos, resultando em 10, ou seja:
  </Text>
  <Text
    style={
      Object {
        "fontSize": 15,
        "marginVertical": 10,
      }
    }
  >
    ((B1 * 2) + (B2 * 2) + (B3 * 3) + (B4 * 3)) / 10
  </Text>
  <Text
    style={
      Object {
        "fontSize": 15,
        "marginVertical": 10,
      }
    }
  >
    Para os cursos superiores, segue a mesma lógica, mas com apenas duas notas bimestrais, com peso total de 5, ou seja:
  </Text>
  <Text
    style={
      Object {
        "fontSize": 15,
        "marginVertical": 10,
      }
    }
  >
    ((B1 * 2) + (B2 * 3)) / 5 
  </Text>
</View>
`;
